Example/AblyChatExample/Mocks/Mocks.swift

```diff
@@ -0,0 +1,177 @@
+import Ably
+import AblyChat
+
+actor MockChatClient: ChatClient {
+    let realtime: RealtimeClient
+    nonisolated let clientOptions: ClientOptions
+    nonisolated let rooms: Rooms
+
+    init(realtime: RealtimeClient, clientOptions: ClientOptions?) {
+        self.realtime = realtime
+        self.clientOptions = clientOptions ?? .init()
+        self.rooms = MockRooms(clientOptions: self.clientOptions)
+    }
+    
+    nonisolated var connection: any Connection {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var clientID: String {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockRooms: Rooms {
+    let clientOptions: ClientOptions
+    
+    func get(roomID: String, options: RoomOptions) async throws -> any Room {
+        MockRoom(roomID: roomID, options: options)
+    }
+    
+    func release(roomID: String) async throws {
+        fatalError("Not yet implemented")
+    }
+    
+    init(clientOptions: ClientOptions) {
+        self.clientOptions = clientOptions
+    }
+}
+
+actor MockRoom: Room {
+    internal nonisolated let roomID: String
+    internal nonisolated let options: RoomOptions
+    
+    internal init(roomID: String, options: RoomOptions) {
+        self.roomID = roomID
+        self.options = options
+    }
+    
+    nonisolated var messages: any Messages {
+        MockMessages(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var presence: any Presence {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var reactions: any RoomReactions {
+        MockRoomReactions(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var typing: any Typing {
+        MockTyping(clientID: "AblyTest", roomID: roomID)
+    }
+
+    nonisolated var occupancy: any Occupancy {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var status: any RoomStatus {
+        fatalError("Not yet implemented")
+    }
+
+    func attach() async throws {
+        fatalError("Not yet implemented")
+    }
+
+    func detach() async throws {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockMessages: Messages {
+    let clientID: String
+    let roomID: String
+    let channel: RealtimeChannel
+    
+    init(clientID: String, roomID: String) {
+        self.clientID = clientID
+        self.roomID = roomID
+        self.channel = MockRealtimeChannel()
+    }
+    
+    func subscribe(bufferingPolicy: BufferingPolicy) async -> MessageSubscription {
+        MessageSubscription(mockAsyncSequence: MockMessageSubscription(clientID: clientID, roomID: roomID)) {_ in 
+            MockMessagesPaginatedResult(clientID: self.clientID, roomID: self.roomID)
+        }
+    }
+    
+    func get(options: QueryOptions) async throws -> any PaginatedResult<Message> {
+        MockMessagesPaginatedResult(clientID: self.clientID, roomID: self.roomID)
+    }
+    
+    func send(params: SendMessageParams) async throws -> Message {
```

Comment:
Are you going to make it so that this is also emitted on the subscription? ditto for reactions

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/18f3099e4b00ee8834006bba0bfe0cda8d5d8169